js/reference.js

```diff
@@ -396,6 +446,29 @@ var ERMrest = (function(module) {
         },
 
         /**
+         * Return a new Reference with the new sorting
+         *
+         * @param {Object[]} sort an array of objects in the format
+         * {"column":columname, "descending":true|false}
+         * in order of priority. Undfined, null or Empty array to use default sorting.
+         */
+        sort: function(sort) {
+
+            // make a Reference copy
+            var newReference = _referenceCopy(this);
+            // change ._sort
+            if (!sort || sort.length === 0) {
+                delete newReference._sort;
+            }
+            else {
+                verify((sort instanceof Array), "input should be an array");
```

Comment:
You could also add a deeper verification here.

```javascript
function isValidElement (element, index, array) {
    return (typeof element == 'object' &&
            typeof element.column == 'string' &&
            typeof element.order == 'string');
}
verify(sort.every(isValidElement), "invalid arguments in array");
```

MDN says that Array.every is supported fairly well.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every